CSharp/Library/Microsoft.Bot.Builder/ConnectorEx/BotData.cs

```diff
@@ -89,18 +91,25 @@ public interface IBotDataStore<T>
     /// </summary>
     public class InMemoryDataStore : IBotDataStore<BotData>
     {
-        internal readonly ConcurrentDictionary<string, string> store = new ConcurrentDictionary<string, string>();
+        internal static readonly MemoryCache store = new MemoryCache(nameof(InMemoryDataStore), new NameValueCollection() { { "PhysicalMemoryLimitPercentage", "50" } });
```

Comment:
>static [](start = 17, length = 6)

Doesn't InMemoryDataStore have lifetime specified in the container?  Making it static interferes with testability.